File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/SeekBarChangeEvent.java
package com.jakewharton.rxbinding2.widget;

import android.support.annotation.NonNull;
import android.widget.SeekBar;

public abstract class SeekBarChangeEvent {
  SeekBarChangeEvent() {
  }

  /** The view from which this event occurred. */
  @NonNull public abstract SeekBar view();
}

File: rxbinding/src/main/java/com/jakewharton/rxbinding4/widget/SeekBarChangeEvent.kt
package com.jakewharton.rxbinding4.widget

import android.widget.SeekBar

sealed class SeekBarChangeEvent {
  /** The view from which this event occurred.  */
  abstract val view: SeekBar
}

data class SeekBarProgressChangeEvent(
  override val view: SeekBar,
  val progress: Int,
  val fromUser: Boolean
) : SeekBarChangeEvent()

data class SeekBarStartChangeEvent(
  override val view: SeekBar
) : SeekBarChangeEvent()

data class SeekBarStopChangeEvent(
  override val view: SeekBar
) : SeekBarChangeEvent()
